src/main/java/hudson/plugins/promoted_builds/conditions/GroovyConditionDescriptor.java

```diff
@@ -4,41 +4,53 @@
 import hudson.PluginManager;
 import hudson.PluginWrapper;
 import hudson.model.AbstractProject;
+import hudson.model.Job;
+import hudson.model.TaskListener;
 import hudson.plugins.promoted_builds.PromotionConditionDescriptor;
+import javafx.concurrent.Task;
 import jenkins.model.Jenkins;
 
+import javax.annotation.Nonnull;
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
 /**
  * Declared outside GroovyCondition as it depends on the optional 'script-security' dependency
  */
 @Extension(optional = true)
-public final class GroovyConditionDescriptor extends PromotionConditionDescriptor {
```

Comment:
why?

Reply:
No specific reason known for this just maybe because of the Job integrity it required multiple descriptors in pipelines(maybe) , just that the code compiled correctly this way would it break sth if final is removed?